pkg/clustersync/clusterSync.go

```diff
@@ -314,4 +324,47 @@ func processClusterDelete(ctx context.Context, obj interface{}) {
 		return
 	}
 	dao.DeleteClusterAndResources(ctx, clusterName, deleteClusterNode)
+
+}
+
+// finds lingering data in database from deleted/detached clusters or clusters with search-collector-addon disabled:
+func confirmDelete(ctx context.Context, dynamicClient dynamic.Interface, gvr schema.GroupVersionResource) ([]string, error) {
```

Comment:
Maybe a better name would be `findLingeringClusterResources()` or `findStaleClusterResources()`